plugin.video.vrt.nu/resources/lib/vrtplayer/metadatacreator.py

```diff
@@ -90,137 +124,115 @@ def season(self, value):
         self._season = value
 
     @property
-    def episode(self):
-        return self._episode
+    def subtitle(self):
+        return self._subtitle
 
-    @episode.setter
-    def episode(self, value):
-        self._episode = value
+    @subtitle.setter
+    def subtitle(self, value):
+        self._subtitle = value
 
     @property
-    def year(self):
-        return self._year
+    def title(self):
+        return self._title
 
-    @year.setter
-    def year(self, value):
-        self._year = value
+    @title.setter
+    def title(self, value):
+        self._title = value
 
     @property
-    def mediatype(self):
-        return self._mediatype
+    def tvshowtitle(self):
+        return self._tvshowtitle
 
-    @mediatype.setter
-    def mediatype(self, value):
-        self._mediatype = value
+    @tvshowtitle.setter
+    def tvshowtitle(self, value):
+        self._tvshowtitle = value
 
     @property
-    def url(self):
-        return self._url
-
-    @url.setter
-    def url(self, value):
-        self._url = value
+    def year(self):
+        return self._year
 
-    @property
-    def brands(self):
-        return self._brands
+    @year.setter
+    def year(self, value):
+        self._year = value
 
-    @brands.setter
-    def brands(self, value):
-        self._brands = value
+    def get_video_dict(self):
+        try:
+            epoch = datetime.fromtimestamp(0, dateutil.tz.tzlocal())
+        except ValueError:
+            epoch = datetime.fromtimestamp(0, dateutil.tz.tzwinlocal())
```

Comment:
This call may raises an `UnicodeEncodeError` and makes opening `A-Z`, `Most recent` and sub-listings impossible to open.

Please let me know if this update should be merged despite that error or if you want to fix it.

Reply:
In what situation would this cause a UnicodeEncodeError? I tried on both Python 2.7 Linux and Python 3.6 Windows.

Reply:
This is the error I get with Python 2 on Windows.
```
EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
     - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
    Error Type: <type 'exceptions.UnicodeEncodeError'>
    Error Contents: 'ascii' codec can't encode character u'\xe4' in position 67: ordinal not in range(128)
    Traceback (most recent call last):
      File "...\Kodi\addons\plugin.video.vrt.nu\addon.py", line 60, in <module>
        router(sys.argv[2][1:])
      File "...\Kodi\addons\plugin.video.vrt.nu\addon.py", line 36, in router
        vrt_player.show_tvshow_menu_items(path=None)
      File "...\Kodi\addons\plugin.video.vrt.nu\resources\lib\vrtplayer\vrtplayer.py", line 62, in show_tvshow_menu_items
        tvshow_items = self._api_helper.get_tvshow_items(path)
      File "...\Kodi\addons\plugin.video.vrt.nu\resources\lib\vrtplayer\vrtapihelper.py", line 59, in get_tvshow_items
        video_dict=metadata.get_video_dict()))
      File "...\Kodi\addons\plugin.video.vrt.nu\resources\lib\vrtplayer\metadatacreator.py", line 162, in get_video_dict
        epoch = datetime.fromtimestamp(0, dateutil.tz.tzwinlocal())
      File "...\Kodi\addons\script.module.dateutil\lib\dateutil\tz\win.py", line 248, in __init__
        with winreg.OpenKey(handle, tzkeyname) as tzkey:
    UnicodeEncodeError: 'ascii' codec can't encode character u'\xe4' in position 67: ordinal not in range(128)
    -->End of Python script error report<--
```
The reason why this happens is that `tzkeyname` is `SOFTWARE\Microsoft\Windows NT\CurrentVersion\Time Zones\Mitteleuropäische Zeit`.

Reply:
And what would be the fix for this? A newer dateutil maybe? Please merge this anyway as I expect no issues for the majority of our users. And the plugin is now broken for everyone. 😉

I'll fix this with some other improvements and do a v1.8.1 release later.

Reply:
https://github.com/dateutil/dateutil/issues/210

Reply:
@Rechi Could you try this patch: https://github.com/pietje666/plugin.video.vrt.nu/pull/156 ?
I cannot easily reproduce this myself, but at least was not affected by other **tzlocal()** instances in our code. Fixing the others would be a lot harder to do.

Reply:
After replacing `datetime.datetime` with `datetime` I was able to open `A-Z`, but there I couldn't open anything.
I got the following exception (also for `Most recent`).
```
EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
     - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
    Error Type: <type 'exceptions.TypeError'>
    Error Contents: can't compare offset-naive and offset-aware datetimes
    Traceback (most recent call last):
      File "...\Kodi\addons\plugin.video.vrt.nu\addon.py", line 60, in <module>
        router(sys.argv[2][1:])
      File "...\Kodi\addons\plugin.video.vrt.nu\addon.py", line 42, in router
        vrt_player.show_episodes(params.get('video_url'))
      File "...\Kodi\addons\plugin.video.vrt.nu\resources\lib\vrtplayer\vrtplayer.py", line 102, in show_episodes
        episode_items, sort, ascending = self._api_helper.get_episode_items(path)
      File "...\Kodi\addons\plugin.video.vrt.nu\resources\lib\vrtplayer\vrtapihelper.py", line 91, in get_episode_items
        episode_items, sort, ascending = self._map_to_episode_items(api_json.get('results', []), path)
      File "...\Kodi\addons\plugin.video.vrt.nu\resources\lib\vrtplayer\vrtapihelper.py", line 201, in _map_to_episode_items
        video_dict=metadata.get_video_dict(),
      File "...\Kodi\addons\plugin.video.vrt.nu\resources\lib\vrtplayer\metadatacreator.py", line 176, in get_video_dict
        if self.ontime and self.ontime != epoch:
    TypeError: can't compare offset-naive and offset-aware datetimes
```

Reply:
Ah, I was expecting that 0 to translate to UTC, better to use dateutil.tz.UTC then.

Reply:
@Rechi I am trying to get this fixed in dateutil for Python 2.7 (getting this fixed in Python 2.7 is unfortunately no longer an option). https://github.com/dateutil/dateutil/pull/912